### PR TITLE
1 -- feat: add AF Team Interaction action type for PM school visits

### PR DESCRIPTION
**Pull request by deepansh96**
## Summary

Adds **AF Team Interaction** — the second enabled action type in the PM school visit system (alongside Classroom Observation). This lets PMs record a structured checklist from their group interaction with teachers at a school.

The form has:
- A **multiselect teacher dropdown** (fetched from the school's teacher list)
- **9 Yes/No questions** organized into 4 sections (Operational Health, Student Performance, Support Needed, Monthly Planning)
- Optional **remarks** on each question
- A **sticky progress bar** showing teacher count and answered questions

Key behavior:
- **Saving** uses lenient validation (partial data is OK)
- **Ending** the action requires all 9 questions answered + at least 1 teacher selected
- AF Team Interaction is **not required** for visit completion — only Classroom Observation is mandatory
- Read-only mode shows teacher names as static text and disables all inputs

---

### New files (8)

| File | What it does |
|------|-------------|
| `src/lib/af-team-interaction.ts` | Config (4 sections, 9 questions), data types, lenient + strict validation functions |
| `src/lib/af-team-interaction.test.ts` | 23 tests covering config integrity, lenient validation (accepts partial, rejects bad types/unknown keys), and strict validation (requires all 9 answers + teachers) |
| `src/lib/teacher-utils.ts` | Shared `Teacher` interface and `getTeacherDisplayName()` helper — extracted from ClassroomObservationForm so both forms can reuse it |
| `src/lib/teacher-utils.test.ts` | 3 tests for display name fallback logic |
| `src/components/visits/AFTeamInteractionForm.tsx` | The form component — teacher checkboxes with Select All/Deselect All, 9 binary questions with remark toggles, progress bar, disabled/read-only mode |
| `src/components/visits/AFTeamInteractionForm.test.tsx` | 22 tests — teacher fetch/toggle, question interaction, gating logic, progress bar, disabled mode, edge cases |
| `docs/LMS Field Visit - Forms and Templates - AF Team Interaction.csv` | Source CSV with the original question list from the product team |
| `docs/ai/af-team-interaction/2026-03-05-af-team-interaction-implementation-plan.md` | Detailed implementation plan (phases, checklists, test specs) |

### Modified source files (8)

| File | What changed |
|------|-------------|
| `src/lib/visit-actions.ts` | Added `af_team_interaction: "AF Team Interaction"` to the `ACTION_TYPES` map (8 → 9 types) |
| `src/components/visits/ClassroomObservationForm.tsx` | Removed local `Teacher` interface and `getTeacherDisplayName` — now imports from shared `teacher-utils.ts` |
| `src/components/visits/ActionTypePickerModal.tsx` | Enabled `af_team_interaction` in the picker (was only `classroom_observation`) |
| `src/components/visits/ActionDetailForm.tsx` | Wired AF Team Interaction form: added config entry, sanitize/bootstrap functions, form renderer dispatch, save-before-end logic, and type-specific error messages |
| `src/components/visits/ActionPointList.tsx` | Added `getAFTeamInteractionStats()` function and renders teacher count + answered question stats on AF team interaction cards |
| `src/app/api/pm/visits/[id]/actions/[actionId]/route.ts` | PATCH route now validates AF team interaction data (lenient for in-progress, strict for completed) |
| `src/app/api/pm/visits/[id]/actions/[actionId]/end/route.ts` | END route now validates AF team interaction data with strict validation (both normal and concurrent-fallback paths) |
| `src/app/api/pm/teachers/route.ts` | Minor whitespace change |

### Modified test files (7)

| File | Tests added | What's tested |
|------|------------|---------------|
| `src/lib/visit-actions.test.ts` | +1 | Updated exhaustiveness check (8 → 9 types) |
| `src/components/visits/ActionTypePickerModal.test.tsx` | +3 | AF team interaction is selectable, submittable, other types still disabled |
| `src/components/visits/ActionPointList.test.tsx` | +13 | AF team interaction stats rendering (4 component tests) + `getAFTeamInteractionStats()` unit tests (9 edge case tests) |
| `src/app/api/pm/visits/[id]/actions/[actionId]/route.test.ts` | +7 | PATCH validation for AF team interaction — lenient accepts empty/partial, rejects bad types; strict rejects incomplete, accepts complete |
| `src/app/api/pm/visits/[id]/actions/[actionId]/end/route.test.ts` | +5 | END validation — rejects incomplete/null/empty-teachers data, succeeds with complete data, concurrent fallback path |
| `src/app/api/pm/visits/[id]/complete/route.test.ts` | +1 | Visit completes with AF team interaction alongside classroom observation (supplementary, not blocking) |
| `src/app/visits/[id]/actions/[actionId]/page.test.tsx` | +5 | AF team interaction renderer loads, payload bootstrap/sanitization, save-before-end flow, error messages |

### E2E changes (3 files)

| File | What changed |
|------|-------------|
| `e2e/fixtures/migrations/20260218190000_add_visit_actions_compat.sql` | Fixed table name from `lms_pm_visit_actions` → `lms_pm_school_visit_actions` (matching production schema) |
| `e2e/helpers/db.ts` | Added `buildCompleteAFTeamInteractionData()` builder; fixed table name in `seedVisitAction()`; added `teacher_id`/`teacher_name`/`grade` to classroom observation builder |
| `e2e/tests/visits.spec.ts` | **5 new E2E tests** + fixes to existing tests (see below) |

#### New E2E tests
1. **PM creates, starts, fills, and ends AF team interaction** — full lifecycle with save-before-end and DB assertions
2. **END blocked when data incomplete** — verifies 422 error and action stays in_progress
3. **Visit cannot complete with only AF team interaction** — confirms classroom observation is still required
4. **Program admin read-only** — no Save/End buttons, static teacher names, disabled radios
5. **Visit completes with both classroom observation and AF team interaction** — both action types coexist

#### E2E fixes to existing tests
- Fixed add/delete test to use `af_team_interaction` (was using disabled `teacher_feedback`)
- Fixed classroom observation test: auto-navigate after Start (removed manual Open click), added teacher/grade selection before rubric fill
- Fixed table name in DB queries (`lms_pm_visit_actions` → `lms_pm_school_visit_actions`)
- Added `reload()` before admin complete-visit assertion
- Seeded test teachers for AF team interaction tests in `beforeAll`

### Documentation (2 files)

| File | What changed |
|------|-------------|
| `CLAUDE.md` | Added AF Team Interaction section, updated test counts (75→78 files, 1142→1228 tests), added new test file entries |
| `docs/ai/project-context.md` | Added AF team interaction data shape/validation docs, updated action type count (8→9), updated enabled types, updated test counts |

---

## Test plan

- [x] All 1228 unit tests pass (`npm run test`)
- [x] All 34 E2E tests pass (`npm run test:e2e`) — 29 existing + 5 new
- [x] TypeScript compiles cleanly (`npx tsc --noEmit`)
- [x] Manual smoke test: create → start → fill → save → end → complete visit flow

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/ai/af-team-interaction/2026-03-05-af-team-interaction-implementation-plan.md [hunk 1]
```diff
@@ -0,0 +1,1016 @@
+# Plan: Add AF Team Interaction Action Type
```
Comment (pritamps) [inline docs/ai/af-team-interaction/2026-03-05-af-team-interaction-implementation-plan.md:1]: If this is done, can we mark the done tasks as done and make it clear that this is now "documentation" and not "plan"?

File: docs/LMS Field Visit - Forms and Templates - AF Team Interaction.csv [hunk 2]
```diff
@@ -0,0 +1,13 @@
+,,,,,,,
```
Comment (pritamps) [inline docs/LMS Field Visit - Forms and Templates - AF Team Interaction.csv:1]: Why is this file here?

File: e2e/tests/visits.spec.ts [hunk 3]
```diff
@@ -476,4 +506,168 @@ test.describe("Visits — Phase 6.3 E2E scenarios", () => {
...
+  test("pm-creates-starts-fills-and-ends-af-team-interaction", async ({ pmPage }) => {
+    const { visitId } = await seedTestVisit(pool, schoolCode);
+
+    await setGoodGps(pmPage);
+    await pmPage.goto(`/visits/${visitId}`);
+
+    // Add AF Team Interaction via picker
+    await pmPage.getByRole("button", { name: "Add Action Point" }).click();
+    const dialog = pmPage.getByRole("dialog");
+    await dialog.getByLabel("AF Team Interaction").click();
```
Comment (pritamps) [inline e2e/tests/visits.spec.ts:750]: Who owns updating tests when for example, this label is updated?

**Review by pritamps — APPROVED** [review]
Approved!

File: e2e/tests/visits.spec.ts (hunk 3 above)
Comment (deepansh96) [reply e2e/tests/visits.spec.ts:750]: I only own. If I update the label later, these tests will fail and I'll update (or get the agent to update or it will update on its own when it sees the test fail)

File: docs/LMS Field Visit - Forms and Templates - AF Team Interaction.csv (hunk 2 above)
Comment (deepansh96) [reply docs/LMS Field Visit - Forms and Templates - AF Team Interaction.csv:1]: this was for context for claude code, a download of the rubric sheet that Vishal and team have created. Will move it to archive

File: docs/ai/af-team-interaction/2026-03-05-af-team-interaction-implementation-plan.md (hunk 1 above)
Comment (deepansh96) [reply docs/ai/af-team-interaction/2026-03-05-af-team-interaction-implementation-plan.md:1]: yeah i'll be moving them to archive folder, forgot that
